the-collab-lab/tcl-24-smart-shopping-list
# 2 am jp router app

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description


This PR adds react-router-dom to the dependencies and three new components: NavBar, AddItem, and List. 

In NavBar we create the application links. We made use of the `<NavLink>` tag from react-router-dom to add styling attributes and display the current link selection in bold.

In App.js we configure the routes of the app, being able to render these different components through their own routes.


## Related Issue
Closes #3 



## Acceptance Criteria



## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before




### After


<img width="944" alt="Screen Shot 2021-04-07 at 2 52 12 PM" src="https://user-images.githubusercontent.com/12903447/113925616-dba51700-97b0-11eb-8a4e-e6c91cb51010.png">


## Testing Steps / QA Criteria


- From your terminal, pull down the branch
- Run npm install to install the dependencies
- Run npm start to view the features in the browser
- Click links
- Check for bold text display
- Check that path matches the selected link.
